cmd/gateway.go

```diff
@@ -0,0 +1,468 @@
+		return err
+	}
+
+	params := gateway.NewServerDeleteParams()
+	params.SetProjectuid(projectUID)
+	params.SetName(name)
+	resp, err := cloud.Gateway.ServerDelete(params, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete gateway server %s/%s: %w", projectUID, name, err)
+	}
+	return output.YAMLPrinter{}.Print(resp.Payload)
+}
+
+func serverDescribe(cmd *cobra.Command, args []string) error {
+	projectUID, name, err := projectUIDAndInstanceName(args)
+	if err != nil {
+		return err
+	}
+	params := gateway.NewServerGetParams()
+	params.SetProjectuid(projectUID)
+	params.SetName(name)
+	resp, err := cloud.Gateway.ServerGet(params, nil)
+	if err != nil {
+		return fmt.Errorf("failed to describe a gateway: %w", err)
+	}
+	return output.YAMLPrinter{}.Print(resp.Payload)
+}
+
+func serverList(cmd *cobra.Command, args []string) error {
+	if len(args) != 0 {
+		return errors.New("no argument allowed")
+	}
+
+	project := viper.GetString(flagProject)
+	if viper.GetBool(flagAllProjects) {
+		if project != "" {
+			return errors.New("only one of two flags allowed")
+		}
+		resp, err := cloud.Gateway.ServerListAll(gateway.NewServerListAllParams(), nil)
+		if err != nil {
+			return fmt.Errorf("listing gateway servers: %w", err)
+		}
+		return output.YAMLPrinter{}.Print(resp.Payload)
+	}
+
+	if project == "" {
+		return errors.New("project missing")
+	}
+	params := gateway.NewServerListParams()
+	params.SetProjectuid(project)
+	resp, err := cloud.Gateway.ServerList(params, nil)
+	if err != nil {
+		return fmt.Errorf("listing gateway servers: %w", err)
+	}
+	return output.YAMLPrinter{}.Print(resp.Payload)
+}
+
+func serverUpdate(cmd *cobra.Command, args []string) error {
+	projectUID, name, err := projectUIDAndInstanceName(args)
+	if err != nil {
+		return err
+	}
+	params := gateway.NewServerPatchParams()
+	params.SetProjectuid(projectUID)
+	params.SetName(name)
+
+	parsedPipes, err := parsePipeSpecs(viper.GetStringSlice(serverUpdateFlagPipesToAdd))
+	if err != nil {
+		return fmt.Errorf("failed to parse flag `pipes`: %w", err)
+	}
+	req := &models.V1GatewayServerPatchRequest{
+		ProjectUID: ptr(projectUID),
+		Name:       ptr(name),
+		Pipes:      parsedPipes,
+	}
+	params.SetBody(req)
+
+	resp, err := cloud.Gateway.ServerPatch(params, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create gateway server: %w", err)
+	}
+	return output.YAMLPrinter{}.Print(resp.Payload)
+}
+
+/* other than handlers */
+
+func defineFlagProject(c *cobra.Command) {
+	c.Flags().String(flagProject, "", "Project UID of gateway instance")
+}
+
+// MIC := multiple-instance-command
+func defineFlagProjectMIC(c *cobra.Command) {
+	// `--project` or `--all-projects`
+	defineFlagProject(c)
+	c.Flags().BoolP(flagAllProjects, "A", false, "Instances of all projects")
+}
+
+func defineRequiredFlagProject(c *cobra.Command) error {
+	defineFlagProject(c)
+	must(c.MarkFlagRequired(flagProject))
+	return nil
+}
+
+func must(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
```

Comment:
Also very general. Maybe a little better located in the helper package. Probably this should be used in all the other places of this package, too.